### PR TITLE
Add tagOverrides and contextObjects

### PR DESCRIPTION
**Pull request by christopheranderson**
Addresses #209 

Also updated documentation.

Didn't add any additional tests since this was a lightweight addition and it didn't look like we had any specific tests for the existing `track` method.

### CODE REVIEW

File: Library/Client.ts [hunk 1]
```diff
@@ -44,8 +44,11 @@ class Client {
      * @param   name    A string to identify this event in the portal.
      * @param   properties  map[string, string] - additional data used to filter events and metrics in the portal. Defaults to empty.
      * @param   measurements    map[string, number] - metrics associated with this event, displayed in Metrics Explorer on the portal. Defaults to empty.
+     * @param   tagOverrides the context tags to use for this telemetry which overwrite default context values
```
Comment (OsvaldoRosado) [inline Library/Client.ts]: This should probably have a listing for contextObjects as well

Comment (christopheranderson) [inline Library/Client.ts]: It probably should. Any thoughts on the right language there? I didn't find any good descriptions of what that contextObjects collection does, though I can take a crack at it, based on my understanding.

Comment (OsvaldoRosado) [inline Library/Client.ts]: Perhaps something like "An event-specific context that will be passed to telemetry processors handling this event before it is sent. For a context spanning your entire operation, consider appInsights.getCorrelationContext"

Comment (christopheranderson) [inline Library/Client.ts]: Done

**Pull request merged**